src/coreclr/src/jit/optcse.cpp

```diff
@@ -459,14 +476,36 @@ unsigned Compiler::optValnumCSE_Index(GenTree* tree, Statement* stmt)
         //
         assert(vnLibNorm == vnStore->VNNormalValue(vnOp2Lib));
     }
-    else // Not a GT_COMMA
+    else if (enableSharedConstCSE && tree->IsIntegralConst())
+    {
+        assert(vnStore->IsVNConstant(vnLibNorm));
+        key = vnStore->CoercedConstantValue<size_t>(vnLibNorm);
+
+        // We don't shared small offset constants when we require a reloc
+        if (!tree->AsIntConCommon()->ImmedValNeedsReloc(this))
+        {
+            // Make constants that have the same upper bits use the same key
+
+            // Shift the key right by CSE_CONST_SHARED_LOW_BITS bits and set the upper bits to zero
+            key >>= CSE_CONST_SHARED_LOW_BITS;
+        }
+        assert((key & TARGET_SIGN_BIT) == 0);
```

Comment:
can it be not null for `ImmedValNeedsReloc(this)`?

Reply:
No it can't be null when we have a reloc.